lib/elm-test.js

```diff
@@ -354,16 +45,89 @@ const parsePositiveInteger = (flag /*: string */) => (
 const parseReport = (flag /*: string */) => (
   string /*: string */
 ) /*: typeof Report.Report */ => {
-  const result = Report.parse(string);
-  switch (result.tag) {
-    case 'Ok':
-      return result.value;
-    case 'Error':
-      console.error(`error: option '${flag}' ${result.error}`);
-      throw process.exit(1);
+  try {
+    return Report.parse(string);
+  } catch (error) {
+    console.error(`error: option '${flag}' ${error.message}`);
+    throw process.exit(1);
   }
 };
 
+function findClosest(
+  dir /*: string */,
+  dirToPath /*: (dir: string) => string */
```

Comment:
We only call this function in one place so there is only ever one `dirToPath` value (which is `ElmJson.getPath`) could we hard code it? I think it would make the purpose of this function clearer and the logic more straightforward.

Reply:
Good idea, I made the change.